src/App.tsx

```diff
@@ -1,12 +1,107 @@
 import React from 'react';
 import './App.scss';
+import { GoodsList } from './components/GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+type State = {
+  goods: Good[];
+  goodsRed: Good[];
+  goods5: Good[];
+  getAllGoods: boolean;
+  getAllRed: boolean;
+  get5: boolean;
+};
+
+export class App extends React.Component<{}, State> {
+  state: State = {
+    goods: [],
+    goodsRed: [],
+    goods5: [],
+    getAllGoods: false,
+    getAllRed: false,
+    get5: false,
```

Comment:
no need to have all these properties. On each button click, you should make a request to the server for loading needed goods